cpp/example/src/process_csv.cpp

```diff
@@ -0,0 +1,69 @@
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+
+#include <cudf/aggregation.hpp>
+#include <cudf/groupby.hpp>
+#include <cudf/io/csv.hpp>
+#include <cudf/table/table.hpp>
+
+std::unique_ptr<cudf::table> read_csv(std::string const& file_path)
+{
+  auto source_info    = cudf::io::source_info(file_path);
+  auto builder        = cudf::io::csv_reader_options::builder(source_info);
+  auto options        = builder.build();
+  auto data_with_meta = cudf::io::read_csv(options);
+  return std::move(data_with_meta.tbl);
+}
+
+void write_csv(cudf::table_view const& tbl_view, std::string const& file_path)
+{
+  auto sink_info = cudf::io::sink_info(file_path);
+  auto builder   = cudf::io::csv_writer_options::builder(sink_info, tbl_view);
+  auto options   = builder.build();
+  cudf::io::write_csv(options);
+}
+
+std::unique_ptr<cudf::table> average_closing_price(cudf::table_view stock_info_table)
+{
+  // Schema: | Company | Date | Open | High | Low | Close | Volume |
+  auto keys = cudf::table_view{{stock_info_table.column(0)}};  // Company
+  auto val  = stock_info_table.column(5);                      // Close
+
+  // Compute the average of each company's closing price with entire column
+  cudf::groupby::groupby grpby_obj(keys);
+  cudf::groupby::aggregation_request agg_request{val, {}};
+  agg_request.aggregations.push_back(cudf::make_mean_aggregation());
+  std::vector<cudf::groupby::aggregation_request> requests;
+  requests.push_back(std::move(agg_request));
```

Comment:
```suggestion
  cudf::groupby::aggregation_request agg_request{val, {cudf::make_mean_aggregation()}};
  std::vector<cudf::groupby::aggregation_request> requests{std::move(agg_request)};
```

Reply:
I tried this, but getting compile error:
```
In file included from /usr/include/c++/9/memory:65,
                 from /workspace/build/_deps/cudf-src/cpp/include/cudf/aggregation.hpp:22,
                 from /workspace/src/process_csv.cpp:1:
/usr/include/c++/9/bits/stl_uninitialized.h: In instantiation of '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = const std::unique_ptr<cudf::aggregation>*; _ForwardIterator = std::unique_ptr<cudf::aggregation>*]':
/usr/include/c++/9/bits/stl_uninitialized.h:307:37:   required from '_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, std::allocator<_Tp>&) [with _InputIterator = const std::unique_ptr<cudf::aggregation>*; _ForwardIterator = std::unique_ptr<cudf::aggregation>*; _Tp = std::unique_ptr<cudf::aggregation>]'
/usr/include/c++/9/bits/stl_vector.h:1582:33:   required from 'void std::vector<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = const std::unique_ptr<cudf::aggregation>*; _Tp = std::unique_ptr<cudf::aggregation>; _Alloc = std::allocator<std::unique_ptr<cudf::aggregation> >]'
/usr/include/c++/9/bits/stl_vector.h:626:2:   required from 'std::vector<_Tp, _Alloc>::vector(std::initializer_list<_Tp>, const allocator_type&) [with _Tp = std::unique_ptr<cudf::aggregation>; _Alloc = std::allocator<std::unique_ptr<cudf::aggregation> >; std::vector<_Tp, _Alloc>::allocator_type = std::allocator<std::unique_ptr<cudf::aggregation> >]'
/workspace/src/process_csv.cpp:41:86:   required from here
/usr/include/c++/9/bits/stl_uninitialized.h:127:72: error: static assertion failed: result type must be constructible from value type of input range
  127 |       static_assert(is_constructible<_ValueType2, decltype(*__first)>::value,
      |                                                                        ^~~~~
CMakeFiles/libcudf_example.dir/build.make:84: recipe for target 'CMakeFiles/libcudf_example.dir/src/process_csv.cpp.o' failed
```

Reply:
Min-repro: https://godbolt.org/z/d6Pqb34ed

Reply:
https://github.com/rapidsai/cudf/blob/500f42c16f19ed475569b9bf2d7055a3abfc25ed/cpp/tests/groupby/groupby_test_util.hpp#L73-L74

Perhaps like this then:

```
std::vector<cudf::groupby::aggregation_request> requests;
requests.emplace_back(val, {cudf::make_mean_aggregation());
```

Reply:
I think there are extra two lines below that are also relevant in creating an aggregation. For it's complexity, I created a separate function `make_single_aggregation_request` to wrap the logic.

An offline discussion mentions all these complexity maybe alleviated once the vectors are replaced with `host_span`. And that APIs like `aggregation_request` shouldn't be owner to an `aggregation_request`.